python_docker/registry.py

```diff
@@ -231,6 +231,10 @@ def pull_image(self, image: str, tag: str = "latest", lazy: bool = False):
         layers.
 
         """
+        # detect official docker images
+        if "/" not in image:
```

Comment:
This is dockerhub specific behavior and I don't want to include it in the default behavior. I'm glad you brought this up though since I've talked with others about having ecr, Digital Ocean registries, azure registries, quay, docker specific providers. I've opened a PR https://github.com/costrouc/python-docker/pull/4 that should address your needs. Just use the `DockerRegistry` provider.